app/src/org/commcare/engine/resource/AndroidResourceManager.java

```diff
@@ -141,7 +143,7 @@ private void loadProfileViaTemp(Resource upgradeProfile)
             UnresolvedResourceException,
             InstallCancelledException {
         tempUpgradeTable.destroy();
-        loadProfileIntoTable(tempUpgradeTable, profileRef);
+        loadRemoteAuthorityProfileIntoTable(tempUpgradeTable, profileRef);
```

Comment:
I think that this should be `loadProfileIntoTable(tempUpgradeTable, profileRef, authority);`. 
If CommCare is closed half way through a local update, then `upgradeTable.getResourceWithId(CommCarePlatform.APP_PROFILE_RESOURCE_ID);` (in calling method) won't be `null`, and hence `loadProfileViaTemp` will be called, meaning that we will try to update from a remote resource.